src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php

```diff
@@ -223,6 +223,26 @@ public function testDumpAsFiles()
         $this->assertStringMatchesFormatFile(self::$fixturesPath.'/php/services9_as_files.txt', $dump);
     }
 
+    public function testNonSharedLazyDumpAsFiles()
+    {
+        $container = include self::$fixturesPath.'/containers/container9.php';
+        $container->getDefinition('bar')->addTag('hot');
+        $container->register('non_shared_foo', \Bar\FooLazyClass::class)
+            ->setFile(realpath(self::$fixturesPath.'/includes/foo_lazy.php'))
+            ->setShared(false)
+            ->setPublic(true)
+            ->setLazy(true);
+        $container->compile();
+        $dumper = new PhpDumper($container);
+        $dump = print_r($dumper->dump(['as_files' => true, 'file' => __DIR__, 'hot_path_tag' => 'hot']), true);
```

Comment:
as_file should be off (there is only one test case where it's enabled, you picked that one :) )
please also don't take container9.php as the base but a much smaller one created for the purpose (see existing ones also)

Reply:
I know that there is only one test case. 
I really need the as_files option enabled otherwise we don't rh issue requierments...

I'll check to use a smaller container file

Reply:
> I really need the as_files option enabled otherwise we don't match the issue requierments...

Oh, right! Looking forward to the smaller fixture then thanks :)

Reply:
ok

Reply:
done